### ISSUE TITLE
flux job: should stdin be ignored in pty mode?

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#3335]
Just noticed this while poking at #3320 :
```
$ flux mini submit -o pty cat
ƒ2tjZqoM
$ echo foo | ./flux job attach  ƒ2tjZqoM
flux-job: client.c:260: pty_client_data: data decode failed: Bad file descriptor
flux-job: client.c:260: pty_client_data: data decode failed: Bad file descriptor
 
^Cflux-job: one more ctrl-C within 2s to cancel or ctrl-Z to detach
[detached: session exiting]
```

Maybe stdin of the attach command just be ignored in the pty case?

**Comment by grondo** [comment]
Is the problem here that your `flux job attach` doesn't have a current terminal but is trying to attach to a remote pty? I don't think `flux job attach` can ignore stdin since that is the fd of the local pty, but perhaps it should error out in this case with `not a terminal`, e.g. as `tmux attach` does:

```
 grondo@asp:~$ echo foo | tmux attach
open terminal failed: not a terminal
```

(I could have sworn this was already done int the pty client code, but I guess it was missed.

**Comment by grondo** [comment]
If this is a valid use-case, maybe `flux job attach` should set the flag `FLUX_PTY_CLIENT_STDIN_PIPE` when stdin isn't a terminal.

**Comment by garlick** [comment]
Right, I guess I was thinking more along the lines of a better error message like the tmux one and a quick exit.

I suppose another question is whether the signal handling (e.g. the double ctrl-C) should be disabled when stdin is not a pty?

**Comment by grondo** [comment]
> I suppose another question is whether the signal handling (e.g. the double ctrl-C) should be disabled when stdin is not a pty?

If `flux job attach` exits immediately when stdin is not a tty is that a problem?

I guess we should take a step back and decide, what should be the behavior if `flux job attach JOBID` finds that JOBID has a pty, but stdin is not a tty? Easiest would be to generate a sensible error rather than the current error, but we could also pipe from stdin as in `flux terminus attach --pipe` if piping input to pty job might be useful (it would be an interesting way to drive tests, see `t0020-terminus.t` for examples)
